### PR TITLE
refactor: update deploying to feature branches

### PR DESCRIPTION
**Pull request by radnov**
This PR updates the "Test, build and deploy" workflow to allow for deploying im-* PRs to a `feature` namespace. The workflow can now be triggered by both pushe and pull_request events and uses the `CLASSIFICATION` and `ENVIRONMENT` variables to distinguish between the separate feature environments.

### CODE REVIEW

**Review by tonsV2 — APPROVED** [review]

**Review by tonsV2 — APPROVED** [review]

File: .github/workflows/im-build-test-deploy.yaml [hunk 1]
```diff
@@ -43,7 +43,8 @@ on:
 
 jobs:
   build:
-    if: github.event_name == 'push'
+    env:
+      DEPLOY_ENVIRONMENT: ${{ github.event_name == 'push' || (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'deploy-feature')) }}
```
Comment (teleivo) [inline .github/workflows/im-build-test-deploy.yaml:47]: I think we should find a label name without `feature` as I might deploy a bug fix. How about `deploy-im` or simply `deploy` if this is also independent of im 🤔

File: .github/workflows/im-build-test-deploy.yaml [hunk 2]
```diff
@@ -96,10 +97,16 @@ jobs:
...
         run: |
-          echo "ENVIRONMENT=feature" >> $GITHUB_ENV
-          echo "CLASSIFICATION=dev" >> $GITHUB_ENV
+          # convert to lower case
+          HEAD_BRANCH_LOWERCASE=${GITHUB_HEAD_REF,,}
+          # substitute all forward slashes "/" with hyphens "-"
+          HEAD_BRANCH_LOWERCASE_NO_SLASHES=${HEAD_BRANCH_LOWERCASE//\//-}
+
+          echo "ENVIRONMENT=$HEAD_BRANCH_LOWERCASE_NO_SLASHES" >> $GITHUB_ENV
+          echo "CLASSIFICATION=feature" >> $GITHUB_ENV
```
Comment (teleivo) [inline .github/workflows/im-build-test-deploy.yaml:108]: I feel the same for classification `feature` as for the label. Would we use this classification for anything other then deployments triggered via github? If not we could name it `github`.

File: .github/workflows/im-build-test-deploy.yaml (hunk 1 above)
Comment (radnov) [reply .github/workflows/im-build-test-deploy.yaml:47]: We're deploying to the `instance-manager-feature` namespace, that's why I chose `deploy-feature`. Maybe just `deploy` 🤷

File: .github/workflows/im-build-test-deploy.yaml (hunk 2 above)
Comment (radnov) [reply .github/workflows/im-build-test-deploy.yaml:108]: The `ENVIRONMENT` and `CLASSIFICATION` variables are used to distinguish between the separate "feature" environments that are deployed within the same `instance-manager-feature` k8s namespace.

Comment (teleivo) [reply .github/workflows/im-build-test-deploy.yaml:108]: I get that, its just that `feature` does not seem right as it does not only related to features, hence my question 🤔 

> Would we use this classification for anything other then deployments triggered via github? If not we could name it github.

Comment (radnov) [reply .github/workflows/im-build-test-deploy.yaml:108]: Yeah, we'd still use that classification for our own personal environments `ivo`, `tons` and `radnov`.

File: .github/workflows/im-build-test-deploy.yaml (hunk 1 above)
Comment (teleivo) [reply .github/workflows/im-build-test-deploy.yaml:47]: @tonsV2 is also in favor of label `deploy` 😄

**Review by teleivo — APPROVED** [review]

**Pull request merged**